### PR TITLE
cmds/core/pci: add a hint that the -s is a glob; remove globals; improve error handling and coverage

### PR DESCRIPTION
**Pull request by rminnich**
Add a simple example too. This may not be enough help, but it is more than we have now.

I think we need a doc.go. This is a very powerful command, in spite of its simplicity, in keeping with its plan 9 inspiration.

### CODE REVIEW

**Review by binjip978 — DISMISSED** [review]

**Comment by binjip978** [comment]
pkg/cpio test failed, but that have nothing to do with this change

panic: Log in goroutine after TestPipeWriteRead has completed: Could not write record "farba": write |1: broken pipe

goroutine 23 [running]:
testing.(*common).logDepth(0xc00013cd00, {0xc0003760c0, 0x35}, 0x3)
	/opt/hostedtoolcache/go/1.22.4/x64/src/testing/testing.go:1028 +0x4c5
testing.(*common).log(...)
	/opt/hostedtoolcache/go/1.22.4/x64/src/testing/testing.go:1010
testing.(*common).Errorf(0xc00013cd00, {0x61f057?, 0x1?}, {0xc00005afa8?, 0x3?, 0x4?})
	/opt/hostedtoolcache/go/1.22.4/x64/src/testing/testing.go:1074 +0x54
github.com/u-root/u-root/pkg/cpio.TestPipeWriteRead.func1()
	/home/runner/work/u-root/u-root/pkg/cpio/newc_test.go:454 +0x1c8
created by github.com/u-root/u-root/pkg/cpio.TestPipeWriteRead in goroutine 22
	/home/runner/work/u-root/u-root/pkg/cpio/newc_test.go:451 +0x47c
FAIL	github.com/u-root/u-root/pkg/cpio	0.048s

**Comment by rminnich** [comment]
that's ok, there's more to fix in this command.

**Comment by rminnich** [comment]
ok, tinygo test works, globals removed, weird JSON decoding discovered and "fixed"

File: pkg/pci/pci_test.go [hunk 1]
```diff
@@ -72,37 +74,35 @@ func TestPCIReadConfigRegister(t *testing.T) {
...
-				if !strings.Contains(got.Error(), tt.errWant) {
-					t.Errorf("ReadConfig() = %q, want to contain: %q", got, tt.errWant)
-				}
-			} else {
-				if vals != tt.valsWant {
-					t.Errorf("ReadConfig() = '%#x', want: '%#x'", vals, tt.valsWant)
-				}
+			vals, got := tt.pci.ReadConfigRegister(tt.offset, tt.size)
+			if !errors.Is(got, tt.err) {
+				t.Errorf("ReadConfig() = got %v, want %v", got, tt.err)
```
Comment (10000TB) [inline pkg/pci/pci_test.go:101]: Should this be combined as `t.Fatalf(... same...)`

Comment (rminnich) [reply pkg/pci/pci_test.go:101]: done, thanks!

File: cmds/core/pci/pci_test.go [hunk 2]
```diff
@@ -216,17 +208,68 @@ func TestRegisters(t *testing.T) {
...
+	if err != nil {
+		t.Log(err)
+	}
+	if err := json.Unmarshal(b, &d); err != nil {
+		t.Log(err)
+	}
+	if err := json.Unmarshal(b, &d); err != nil {
+		t.Log(err)
+	}
+	if true {
```
Comment (binjip978) [inline cmds/core/pci/pci_test.go:269]: This really look weird, but why `if true`?

File: cmds/core/pci/pci.go [hunk 3]
```diff
@@ -20,25 +20,55 @@ package main
...
+	verbosity int
+	hexdump   int
+	readJSON  string
+	flags     *flag.FlagSet
+}
+
+// ErrBadJSON is for any JSON Unmarshal error.
+// The JSON package predates wrapped errors. This error is a placeholder for all
+// "something went wrong" JSON parsing errors.
+var ErrBadJSON = errors.New("JSON parsing failed")
```
Comment (binjip978) [inline cmds/core/pci/pci.go:51]: nit: probably no need to export this, it's a main package.

Comment (rminnich) [reply cmds/core/pci/pci.go:51]: ok.

File: cmds/core/pci/pci_test.go (hunk 2 above)
Comment (rminnich) [reply cmds/core/pci/pci_test.go:269]: it is a way to mark how to exercise the bug I'm seeing json, please see the comment. I'd like to leave it there for that reason, until I can find out what's going on. I'm still working on a repro

**Review by binjip978 — APPROVED** [review]

**Pull request merged**